p2p/peer.py

```diff
@@ -608,28 +608,46 @@ def request_block_headers(self, block_number_or_hash: Union[int, bytes],
         self.sub_proto.send_get_block_headers(block_number_or_hash, max_headers, reverse)
 
 
-class PeerPoolSubscriber(ABC):
+class PeerSubscriber(ABC):
     _msg_queue: 'asyncio.Queue[PEER_MSG_TYPE]' = None
 
+    @property
+    @abstractmethod
```

Comment:
> slightly less verbose subclassing

Meaning the subclass doesn't have to add the `@property` decorator?

> I've been a fan of the following pattern.

I'm not fan. I found it sort of confusing when reading the `Transaction` subclasses classes, because we define `get_intrinsic_gas` in the subclass, but then callers would just use `intrinsic_gas`. Paranoid me feels the need to go back and make sure that the superclass defines it to be exactly the same, and make sure that one of the three (caller, subclass, superclass) doesn't just have a bug. If they are exactly the same, then why have both options?

Reply:
That is fair.  I'd consider those concerns reason enough to abandon the pattern.